### PR TITLE
feat: add UnionFind are_connected function to test if two components are connected

### PR DESCRIPTION
**Pull request by yulianka**

### CODE REVIEW

File: workspaces/adventure-pack/goodies/python3/src/union_find/test.py [hunk 1]
```diff
@@ -36,3 +36,13 @@ def test_union_no_work() -> None:
...
     assert all(uf.find(component) == component for component in range(5))
+
+
+def test_are_connected() -> None:
+    uf = UnionFind(8)
+    uf.union(2, 3)
+    uf.union(3, 4)
+
+    assert not uf.are_connected(1, 3)
+    assert uf.are_connected(2, 3)
```
Comment (miorel) [inline workspaces/adventure-pack/goodies/python3/src/union_find/test.py:47]: What do you think about checking if they are connected before and after the union?

File: workspaces/adventure-pack/goodies/python3/src/union_find/test.py [hunk 2]
```diff
@@ -36,3 +36,13 @@ def test_union_no_work() -> None:
...
+    assert uf.are_connected(2, 4)
```
Comment (miorel) [inline workspaces/adventure-pack/goodies/python3/src/union_find/test.py:48]: I like that we are checking the indirect connection as well, thanks!

**Review by miorel — APPROVED** [review]
Looks good!

File: workspaces/adventure-pack/goodies/python3/src/union_find/test.py (hunk 1 above)
Comment (yulianka) [reply workspaces/adventure-pack/goodies/python3/src/union_find/test.py:49]: Thank you for the idea! Done!

**Pull request merged**